pkg/api/query/grpc.go

```diff
@@ -24,6 +24,7 @@ type GRPCAPI struct {
 	replicaLabels               []string
 	queryableCreate             query.QueryableCreator
 	queryEngine                 v1.QueryEngine
```

Comment:
Maybe let's call this `prometheusEngine`? Perhaps we could even put this collection of engines behind another struct like:

```go
type QueryEngineFactory struct {
...
}

func (f *QueryEngineFactory) GetPrometheusEngine(...) v1.QueryEngine {
}

func (f *QueryEngineFactory) GetThanosEngine(...) v1.QueryEngine {
}
```

With this, we wouldn't even have to create those engines in advance. We could create them on the first call to `GetXEngine` :thinking: